opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -311,3 +311,57 @@ def _convert_aggregation_temporality(
             aggregation_temporality=aggregation_temporality,
         )
     return None
+
+
+class _AggregationFactory(ABC):
+    @abstractmethod
+    def create_aggregation(self) -> Aggregation:
+        """Creates an aggregation"""
+
+
+class ExplicitBucketHistogramAggregationFactory(_AggregationFactory):
```

Comment:
```suggestion
class ExplicitBucketHistogramAggregation(_AggregationFactory):
```
The intention of this approach is to expose to the user only what is in the spec. Let's use the spec defined names and save us from having a bigger API.

Reply:
At the same time, this PR should rename current aggregation implementations to make them private in order to separate what is exposed to the user and what is not.

Reply:
I don't think we should mark things private in python modules and then import them in other modules. Can we address in #2389? My plan is to just explicitly re-import public things into `__init__.py` and other top level files and use `__all__` to make it explicit. Everything else will be left in a "private" modules or packages.

Reply:
Sorry to be clear, I'm referring to "this PR should rename current aggregation implementations to make them private".

I am think the suggested change is fine